### PR TITLE
[SM]: Update claim start UI page as per new wireframe

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/config/FrontendAppConfig.scala [hunk 1]
```diff
@@ -61,4 +61,6 @@ class FrontendAppConfig @Inject()(configuration: Configuration) {
   lazy val whitelistDestination: String = configuration.get[String]("filters.whitelist.destination")
   lazy val whitelistExcluded: String = configuration.get[String]("filters.whitelist.excluded")
```
Comment (mikail-khan) [inline app/config/FrontendAppConfig.scala:62]: we are not allowed to use `whitelist` anymore due to Black History month - these need to be changed Whitelist --> Allowlist and Blacklist --> Denylist

File: app/controllers/ClaimPeriodStartController.scala [hunk 2]
```diff
@@ -36,11 +37,10 @@ class ClaimPeriodStartController @Inject()(
   navigator: Navigator,
   getSession: GetSessionAction,
   getData: DataRetrievalAction,
-  requireData: DataRequiredAction,
   formProvider: ClaimPeriodStartFormProvider,
```
Comment (mikail-khan) [inline app/controllers/ClaimPeriodStartController.scala:40]: `ClaimPeriodStartController` should be `ClaimPeriodController`

File: app/controllers/ClaimPeriodStartController.scala [hunk 3]
```diff
@@ -36,11 +37,10 @@ class ClaimPeriodStartController @Inject()(
...
   val controllerComponents: MessagesControllerComponents,
   view: ClaimPeriodStartView
```
Comment (mikail-khan) [inline app/controllers/ClaimPeriodStartController.scala:42]: same

File: app/forms/ClaimPeriodStartFormProvider.scala [hunk 4]
```diff
@@ -16,21 +16,16 @@
...
 
 import forms.mappings.Mappings
-import javax.inject.Inject
 import play.api.data.Form
+import models.ClaimPeriodStart
 
 class ClaimPeriodStartFormProvider @Inject() extends Mappings {
 
-  def apply(): Form[LocalDate] =
+  def apply(): Form[ClaimPeriodStart] =
```
Comment (mikail-khan) [inline app/forms/ClaimPeriodStartFormProvider.scala:27]: same

File: app/config/FrontendAppConfig.scala (hunk 1 above)
Comment (sureshhmrc) [reply app/config/FrontendAppConfig.scala]: removed whitelisting altogether - as we don't need it...

File: app/controllers/ClaimPeriodStartController.scala (hunk 2 above)
Comment (sureshhmrc) [reply app/controllers/ClaimPeriodStartController.scala]: ah , missed these, but just renamed all classes and changes pushed

File: app/models/ClaimPeriod.scala [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    Jun2021,
+    Jul2021,
+    Aug2021,
+    Sep2021,
+    Oct2021,
+    Nov2021,
+    Dec2021
+  )
+
+  def options(form: Form[_], schemeEnds: String, currentMonthYear: YearMonth = YearMonth.now())(
```
Comment (mikail-khan) [inline app/models/ClaimPeriod.scala:67]: what is this `schemeEnds` concept? we don't have such a notion at the moment.

Comment (sureshhmrc) [reply app/models/ClaimPeriod.scala:67]: its the month this JSS scheme ends , and its better if its a configurable value as they may extend scheme..
we did the same in JRS

File: migrations/applied_migrations [hunk 6]
```diff
@@ -1,36 +1,35 @@
...
 echo ""
-echo "Applying migration ClaimPeriodStart"
+echo "Applying migration claimPeriod"
 
 echo "Adding routes to conf/app.routes"
 
 echo "" >> ../conf/app.routes
-echo "GET        /claimPeriodStart                  controllers.ClaimPeriodStartController.onPageLoad(mode: Mode = NormalMode)" >> ../conf/app.routes
-echo "POST       /claimPeriodStart                  controllers.ClaimPeriodStartController.onSubmit(mode: Mode = NormalMode)" >> ../conf/app.routes
+echo "GET        /claimPeriod                        controllers.claimPeriodController.onPageLoad(mode: Mode = NormalMode)" >> ../conf/app.routes
```
Comment (mikail-khan) [inline migrations/applied_migrations:9]: we don't camel case endpoints.

`/claimPeriod` should be `/claim-period`

File: app/models/ClaimPeriod.scala (hunk 5 above)
Comment (sureshhmrc) [reply app/models/ClaimPeriod.scala:67]: its configured in application.conf `schemeEnds = "April 2021"`

File: migrations/applied_migrations (hunk 6 above)
Comment (sureshhmrc) [reply migrations/applied_migrations:9]: you can ignore this file and actual routes have been updated accordingly

**Review by mikail-khan — APPROVED** [review]

**Pull request merged**